### ISSUE TITLE
CheckResultSet inside lamda

### ISSUE THREAD

**Issue by enexusde** [pmd/pmd#4122]
**Affects PMD Version:** 
3.19.0 (mvn pmd:check)
**Rule:**
CheckResultSet

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#checkresultset>

**Description:**

I have this code:

```
		boolean successfullyFound = invoke(schema, (s, d, errs, c) -> {
			ResultSet created = c.getMetaData().getTables(null, null, freeTableName, null);
			boolean missing = !created.next();
			schemaMod.publish(verifyResultId, x -> x.getAddedTableBasically(), x -> x.setAddedTableBasically(!missing));
		}, verify);
```

It looks in the database if a tablename is free to use and not aquired by a table having the same name. If no table having the tablename is found its published as free-to-use.

**Code Sample demonstrating the issue:**

```
boolean successfullyFound = invoke(schema, (s, d, errs, c) -> {
			ResultSet created = c.getMetaData().getTables(null, null, freeTableName, null);
			boolean missing = !created.next(); <<Always check the return of one of the navigation method (next,previous,first,last) of a ResultSet.
			schemaMod.publish(verifyResultId, x -> x.getAddedTableBasically(), x -> x.setAddedTableBasically(!missing));
		}, verify);
```

**Expected outcome:**

PMD reports a violation at line ..., but that's wrong. That's a false positive.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
Maven

**Comment by flyhard** [comment]
Could you provide the failure in a simple class? Lots of variables in your code have unknown type, and it would be nice to get it right the first time around.

**Comment by adangel** [comment]
FYI, I've updated the code sample with a hopefully easier to grasp reproducing sample.

**Comment by flyhard** [comment]
I see how it is triggered, but not yet how to solve it. The check is if the created.next() is in a return-, if- or while-statement.
But since you extracted the result and used it twice, it reports a violation. I am not yet sure on how to fix it.

**Comment by enexusde** [comment]
This time you see how it is triggered, is the Caption of this Issue specific? Since lambda does not take part in your explination, but lambda is part of the issue-title it might be improved.

I can not tell why multiple usages of the next()-result is bad. What I understand is that not-using at-all of the next()-result is bad. Could we turn the trigger into:
- If the next()-result is used in a variable <-- new
- If the next()-result is used by return/if/while <-- old
If, however, the field "missing" is not used would trigger a different rule (like UnusedVariables).

**Comment by jsotuyod** [comment]
The issue is still reproducible under PMD 7.0.0.

However, the issue has no relation to the lambda, but to the usage of the `.next()` check in a variable instead of within a control statement / return statement.

I'm updating the issue's description accordingly with a simpler sample that reproduces the issue.
